MIGRATION.md

```diff
@@ -0,0 +1,90 @@
+# Migrating to the `cloudant-java-sdk` library
+This document is to assist in migrating from the `java-cloudant` (package: `com.cloudant.cloudant-client`) to the newly supported [`cloudant-java-sdk`](https://github.com/IBM/cloudant-java-sdk) (package: `com.ibm.cloud.cloudant`).
+
+## Initializing the client connection
+There are several ways to create a client connection in `cloudant-java-sdk`:
+1. [Environment variables](https://github.com/IBM/cloudant-java-sdk#authentication-with-environment-variables)
+2. [External configuration file](https://github.com/IBM/cloudant-java-sdk#authentication-with-external-configuration)
+3. [Programmatically](https://github.com/IBM/cloudant-java-sdk#programmatic-authentication)
+
+[See the README](https://github.com/IBM/cloudant-java-sdk#code-examples) for code examples on using environment variables.
+
+## Other differences
+1. Fetching the Database object first before performing additional operations is not required. For example, in the case of updating a document you would first call `getDocument` to fetch and then `putDocument` to update.
+2. Model classes are used instead of POJOs.
+3. Sending and receiving plain objects and streaming are both available in some cases. E.g. the `db.find` equivalent `getDocument` is returning with a `Document` object, meanwhile `getDocumentAsStream` is returning with an `InputStream`. The `...AsStream` alternatives can be used next to the `postAllDocs`, `postChanges`, `postSearch`, `postFind`, `postPartitionAllDocs`, `postPartitionSearch`, `postPartitionView`, `postPartitionFind`, and the getDocument  methods.
```

Comment:
I think we can point to what we've already written in the README.  @mojito317 @ricellis wdyt?
```suggestion
3. Sending and receiving byte responses is available for some of our operations.  See [the Raw IO section](https://github.com/IBM/cloudant-java-sdk#raw-io) of `cloudant-java-sdk` README for more details.
```

Reply:
https://github.com/cloudant/java-cloudant/pull/518/commits/aceb42aed9c3ed3a8b07564511934103f899cd1e